src/core/defcomp.jl

```diff
@@ -46,29 +46,37 @@ function _replace_dots(ex)
 end
 
 function _generate_run_func(module_name, comp_name, args, body)
-
     if length(args) != 4
         error("Can't generate run_timestep; requires 4 arguments but got $args")
     end
 
+    (p, v, d, t) = args
+
+    if @capture(t, tname_::ttype_)
+        if ttype == :Timestep
+            ttype = :(Mimi.Timestep)
+        end
+        t = tname
+    else
+        ttype = :Int
+    end
+
     # replace each expression with its dot-replaced equivalent
     body = [MacroTools.prewalk(_replace_dots, expr) for expr in body]
 
-    # add types to the parameters  
-    (p, v, d, t) = args
+    # Generate unique function name for each component so we can store a function pointer.
+    # (At least in julia 0.6, methods cannot be invoked directly.)
+    funcname = Symbol("run_timestep_$(module_name)_$(comp_name)")
```

Comment:
We could probably use an anonymous function here? That way we wouldn't "pollute" the namespace? Can be done later, though.

Reply:
The advantage I see of leaving it this way is that one can type `run_timestep` and hit `TAB` in the REPL and see all the functions, which is potentially useful for developers. This form of pollution has no deleterious effects I'm aware of...